### PR TITLE
Refactor AB

### LINKED ISSUE #446: Refactor com2AB
The current results of com2AB are all over the place, some in dict `resAB` which are a bunch of arrays with no clear indication what the x and y values denote, some only after processABresults is run, after which resAB is augumented, but then split in dicts per profile.... Eg. why is there an x, y and z in the `avaPath` variable, what do they denote?

So I suggest
:
Do the `processABresults` always and have a way clearer structure of the `resAB` dict:
- some genereal Info (e.g. sks, settings etc at the beginning, )
- then a subdict for each profile 
- clear labels in the profile dict

### LINKED ISSUE #560: Address savefigFormat TODO in outAB

### PR DESCRIPTION
**Pull request by matthiasto**
- fix #446 
- fix #560 

all results are now in ResAB which is a dictionary with one subdict for each line analyzed.
the subdicts correspond to the line profile (avaProfile) with x, y, z, s coordinates , the name of the profile and the indexes of the split, alpha, beta... points as well as the sks info.
The old 'postprocess AB' has been moved to com2AB file. Figures are now saved in the standard way (plotUtils function)

### CODE REVIEW

File: avaframe/com2AB/com2ABCfg.ini [hunk 1]
```diff
@@ -24,23 +24,18 @@ distance = 10
 # dsMin meters after the beta point also have an angle bellow 10°
 dsMin = 30
 
+# Optionnal (if left blanc, the avalanche path is read from Inputs/LINES/*AB*.shp and
```
Comment (fso42) [inline avaframe/com2AB/com2ABCfg.ini:27]:
```suggestion
# Optional (if left blank, the avalanche path is read from Inputs/LINES/*AB*.shp and
```

File: avaframe/com2AB/com2ABCfg.ini [hunk 2]
```diff
@@ -24,23 +24,18 @@ distance = 10
...
+# the splitPoint from Inputs/POINTS/*.shp)
+# provide the system path to the avalanche path (relative path where . is avaframe)
```
Comment (fso42) [inline avaframe/com2AB/com2ABCfg.ini:29]: Would also work with a full path, right?

File: avaframe/com3Hybrid/com3Hybrid.py [hunk 3]
```diff
@@ -71,12 +70,12 @@ def maincom3Hybrid(cfgMain, cfgHybrid):
         # make a copy because extendDFAPathKernel might modify avaProfileMass
         avaProfileMassExt = DFAPath.extendDFAPathKernel(cfgHybrid['PATH'], avaProfileMass.copy(), dem, particlesList[0])
         # save profile as AB profile in Inputs
-        pathAB = pathlib.Path(avalancheDir, 'Inputs', 'LINES', 'pathAB_aimec')
-        name = 'massAvaPath'
+        # pathAB = pathlib.Path(avalancheDir, 'Inputs', 'LINES', 'hybrid', 'pathAB_aimec')
+        # name = 'massAvaPath'
```
Comment (fso42) [inline avaframe/com3Hybrid/com3Hybrid.py:74]: remove

File: avaframe/com3Hybrid/com3Hybrid.py [hunk 4]
```diff
@@ -71,12 +70,12 @@ def maincom3Hybrid(cfgMain, cfgHybrid):
...
         avaProfileMassExtOri = copy.deepcopy(avaProfileMassExt)
         avaProfileMassExtOri['x'] = avaProfileMassExtOri['x'] + demOri['header']['xllcenter']
         avaProfileMassExtOri['y'] = avaProfileMassExtOri['y'] + demOri['header']['yllcenter']
-        shpConv.writeLine2SHPfile(avaProfileMassExtOri, name, pathAB)
+        # shpConv.writeLine2SHPfile(avaProfileMassExtOri, name, pathAB)
```
Comment (fso42) [inline avaframe/com3Hybrid/com3Hybrid.py:78]: remove

File: avaframe/com3Hybrid/hybridModel_com1DFACfg.ini [hunk 5]
```diff
@@ -92,7 +92,7 @@ explicitFriction = 1
 frictModel = Coulomb
 #+++++++++++++General Friction parameters
 # tan of bed friction angle used for: samosAT, Coulomb, Voellmy
-mu = 0.510208802718046
+mu = 0.40375846095054385
```
Comment (fso42) [inline avaframe/com3Hybrid/hybridModel_com1DFACfg.ini:95]: Should we limit this to f.3 or f.2?

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/out3Plot/outAB.py [hunk 6]
```diff
@@ -97,154 +92,86 @@ def readABresults(saveOutPath, name, flags):
...
-        name = NameAva[i]
-        resAB = processABresults(resAB, name)
+    saveOutPath = pathDict['saveOutPath']
+    flags = cfgMain['FLAGS']
+    FileNamePlot_ext = [None] * len(resAB.keys())
+    FileNameWrite_ext = [None] * len(resAB.keys())
+    for i, name in enumerate(resAB):
+        avaProfile = resAB[name]
         # Plot the whole profile with beta, alpha ... points and lines
         # TODO: use savefigFormat
```
Comment (fso42) [inline avaframe/out3Plot/outAB.py:116]: remove this then

**Review by fso42 — CHANGES_REQUESTED** [review]

File: avaframe/com2AB/com2ABCfg.ini (hunk 2 above)
Comment (matthiasto) [reply avaframe/com2AB/com2ABCfg.ini:29]: not sure about that, should I try, should it work with the full path?

File: avaframe/com3Hybrid/hybridModel_com1DFACfg.ini (hunk 5 above)
Comment (matthiasto) [reply avaframe/com3Hybrid/hybridModel_com1DFACfg.ini:95]: this is computed inside the algorithm and updated automatically...

File: avaframe/com2AB/com2ABCfg.ini (hunk 2 above)
Comment (matthiasto) [reply avaframe/com2AB/com2ABCfg.ini:29]: Full path works too

**Review by fso42 — APPROVED** [review]

**Pull request merged**